src/sql/SqlStatement.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Long from 'long';
+
+export enum SqlExpectedResultType {
+    ANY,
+    ROWS,
+    UPDATE_COUNT
+}
+
+export type SqlExpectedResultTypeStrings = keyof typeof SqlExpectedResultType;
+
+/**
+ * Options used to change behavior of an SQL query.
+ */
+export interface SqlStatementOptions {
+    /**
+     * The schema name. The engine will try to resolve the non-qualified object identifiers from the statement in the
+     * given schema. If not found, the default search path will be used, which looks for objects in the predefined
+     * schemas `partitioned` and `public`.
+     *
+     * * The schema name is case sensitive. For example, `foo` and `Foo` are different schemas.
+     * * The default value is `null` meaning only the default search path is used.
+     */
+    schema?: string;
+    /**
+     * The execution timeout in milliseconds. If the timeout is reached for a running statement, it will be cancelled forcefully.
+     * Zero value means no timeout. -1 means that the timeout in server config will be used. Other negative values are prohibited.
+     * Defaults to `-1`.
+     */
+    timeoutMillis?: Long;
```

Comment:
Should we allow numbers here too, to make it easier to use?

Reply:
ok it would be better I think